src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/ReleasePrisonerService.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.hmpps.prison.service.enteringandleaving
+
+import org.springframework.data.repository.findByIdOrNull
+import uk.gov.justice.hmpps.prison.api.model.InmateDetail
+import uk.gov.justice.hmpps.prison.api.model.RequestToReleasePrisoner
+import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyLocation
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderBooking
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyLocationRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.hmpps.prison.service.BadRequestException
+import uk.gov.justice.hmpps.prison.service.EntityNotFoundException
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderTransformer
+import kotlin.Result.Companion.failure
+import kotlin.Result.Companion.success
+
+class ReleasePrisonerService(
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val agencyLocationRepository: AgencyLocationRepository,
+  private val externalMovementService: ExternalMovementService,
+  private val transformer: OffenderTransformer,
+  private val caseNoteMovementService: CaseNoteMovementService,
+  private val bedAssignmentMovementService: BedAssignmentMovementService,
+  private val sentenceMovementService: SentenceMovementService,
+  private val paymentsMovementService: PaymentsMovementService,
+  private val activityMovementService: ActivityMovementService,
+) {
+
+  fun releasePrisoner(prisonerIdentifier: String, request: RequestToReleasePrisoner): InmateDetail {
+    val booking = getOffenderBooking(prisonerIdentifier).flatMap { it.isActiveIn() }.getOrThrow()
+    val toLocation = getAgencyLocation(request.toLocationCode).getOrThrow()
+
+    val movement = externalMovementService.updateMovementsForRelease(request.releaseTime, request.movementReasonCode, booking, toLocation, request.commentText)
+
+    caseNoteMovementService.createReleaseNote(booking, movement)
+    bedAssignmentMovementService.endBedHistory(booking.bookingId, movement.movementTime)
+    sentenceMovementService.deactivateSentences(booking.bookingId)
+    paymentsMovementService.endPaymentRules(booking.bookingId, movement.movementDate)
+    activityMovementService.endActivitiesAndWaitlist(booking, toLocation, movement.movementDate, request.movementReasonCode)
+
+    return transformer.transform(booking)
+  }
```

Comment:
The main change is to implement the new releasePrisoner service - not plumbed into the endpoint yet